src/api/app/models/package.rb

```diff
@@ -1251,7 +1251,7 @@ def self.is_binary_file?(filename)
   def serviceinfo
     begin
       dir = Directory.hashed(project: project.name, package: name)
-      return dir['serviceinfo'] if dir
+      return dir['serviceinfo'] || {} if dir
```

Comment:
perhaps ``dir.fetch('serviceinfo', {})`` is a little easier to read in this context?  I find the || combined with the if a little hard - but nitpicking